practical/jython/analyse_particles_for_another_user.jy

```diff
@@ -125,56 +125,25 @@ def switch_security_context(ctx, target_user):
     return ctx
 
 
-def open_omero_image(ctx, image_id):
-    "Convert omero Image object as ImageJ ImagePlus object."
-    browse = gateway.getFacility(BrowseFacility)
-    image = browse.getImage(ctx, long(image_id))
-    pixels = image.getDefaultPixels()
-    sizeZ = pixels.getSizeZ()
-    sizeT = pixels.getSizeT()
-    sizeC = pixels.getSizeC()
-    sizeX = pixels.getSizeX()
-    sizeY = pixels.getSizeY()
-    pixtype = pixels.getPixelType()
-    pixType = FormatTools.pixelTypeFromString(pixtype)
-    bpp = FormatTools.getBytesPerPixel(pixType)
-    isSigned = FormatTools.isSigned(pixType)
-    isFloat = FormatTools.isFloatingPoint(pixType)
-    isLittle = False
-    interleave = False
-
-    store = gateway.getPixelsStore(ctx)
-    pixelsId = pixels.getId()
-    store.setPixelsId(pixelsId, False)
-    stack = ImageStack(sizeX, sizeY)
-    for t in range(0, sizeT):
-        for z in range(0, sizeZ):
-            for c in range(0, sizeC):
-                plane = store.getPlane(z, c, t)
-
-                ImageTools.splitChannels(plane, 0, 1, bpp, False, interleave)
-                pixels = DataTools.makeDataArray(plane, bpp, isFloat, isLittle)
-
-                q = pixels
-                if (len(plane) != sizeX*sizeY):
-                    tmp = q
-                    q = zeros(sizeX*sizeY, 'h')
-                    System.arraycopy(tmp, 0, q, 0, Math.min(len(q), len(tmp)))
-                    if isSigned:
-                        q = DataTools.makeSigned(q)
-
-                if q.typecode == 'b':
-                    ip = ByteProcessor(sizeX, sizeY, q, None)
-                elif q.typecode == 'h':
-                    ip = ShortProcessor(sizeX, sizeY, q, None)
-                stack.addSlice('', ip)
-    # Do something
-    image_name = image.getName() + '--OMERO ID:' + str(image.getId())
-    imp = ImagePlus(image_name, stack)
-    imp.setDimensions(sizeC, sizeZ, sizeT)
-    imp.setOpenAsHyperStack(True)
-    imp.show()
-    return imp
+def open_image_plus(HOST, USERNAME, PASSWORD, PORT, group_id, image_id):
+    "Open the image using the Bio-Formats Importer"
+
+    options = ""
+    options += "location=[OMERO] open=[omero:server="
+    options += HOST
+    options += "\nuser="
+    options += USERNAME
+    options += "\nport="
+    options += str(PORT)
+    options += "\npass="
+    options += PASSWORD
+    options += "\ngroupID="
+    options += str(group_id)
+    options += "\niid="
+    options += str(image_id)
+    options += "]"
+    options += " windowless=true "
```

Comment:
maybe far more readable as a here document with `.format` called on it

Reply:
Sorry @mtbc I'm not sure I understand exactly.

Reply:
```
"""location =[OMERO] open=[omero:server=%s
user=%s
port=%s
pass=%s
groupID=%s
iid=%s] windowless=true""".format(HOST, USERNAME, PORT, PASSWORD, group_id, image_id)
```